### ISSUE TITLE
Autocorrect doesn't change file!?

### ISSUE THREAD

**Issue by sleekweasel** [bbatsov/rubocop#2025]
This file:

```
class BasicInfoCommon < Calabash::ABase
  def await(wait_opts={})
    wait_for_elements_exist([ @@gender_male, @@gender_female], wait_opts, timeout: 5)
    wait_for(timeout:2) {query(@@dob, :text).first != nil}
  end
end
```

sees autocorrect say this:

```
$ rubocop --auto-correct rubo.rb 
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.6-compliant syntax, but you are running 2.1.5.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Inspecting 1 file
C

Offenses:

rubo.rb:2:22: C: [Corrected] Surrounding space missing in default value assignment.
  def await(wait_opts={})
                     ^
rubo.rb:3:30: C: [Corrected] Space inside square brackets detected.
    wait_for_elements_exist([ @@gender_male, @@gender_female], wait_opts, timeout: 5)
                             ^
rubo.rb:4:52: C: [Corrected] Prefer !expression.nil? over expression != nil.
    wait_for(timeout:2) {query(@@dob, :text).first != nil}
                                                   ^^

1 file inspected, 3 offenses detected, 3 offenses corrected
```

but rubocop doesn't actually change the file at all! I can run the --auto-correct command over and over, with the same messages from rubocop.

```
rubocop -V
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.6-compliant syntax, but you are running 2.1.5.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
0.32.0 (using Parser 2.2.2.6, running on ruby 2.1.5 x86_64-darwin14.0)
```

**Comment by sleekweasel** [comment]
Curiously, if I don't put  `wait_opts` in the 'wait_for_element_exist` parameters, it *does* prepend the underscore to the unused parameter, but doesn't fix anything else.

**Comment by jonas054** [comment]
This seems to be fixed in `0.32.1` and as far as I can see it was #1965 that fixed it. It's a bit of a mystery to me why exactly, but there it is. /cc @rrosenblum

**Comment by rrosenblum** [comment]
@jonas054 is correct, this was fixed in `0.32.1`. It should be easy to upgrade to that version.

In this case, RuboCop is trying to correct `query(@@dob, :text).first != nil` to `"query(@@dob, :text).first != nil"` (the exact same thing). The fix was to not trigger a correction if the code is the same. Returning from the corrector lambda before a correction is made seems to also cause this issue.

We may want to look into adjusting the code that determines the how to autocorrect since it isn't able to here.
```ruby
new_code =
  if include_semantic_changes?
    expr.sub(/\s*!=\s*nil/, '') 
  else
    expr.sub(/^(\S*)\s*!=\s*nil/, '!\1.nil?')
  end 
```

**Comment by rrosenblum** [comment]
Actually, this would be corrected by `IncludeSemanticChanges`.
